### ISSUE TITLE
proc_open: Method not implemented

### ISSUE THREAD

**Issue by tverweij** [iolevel/peachpie-vs#2]
I was testing Peachpie and found (in the first project I tried) that a call to proc_open causes a System.NotImplementedException: 'The method or operation is not implemented.'.

**Comment by jakubmisek** [comment]
@tverweij there are a few not implemented code paths - may I ask for the callstack and arguments of the proc_open() function call?

From what I see, `descriptor` of type resource or "file" is not yet implemented.

**Comment by tverweij** [comment]
I was testing this code: https://sourceforge.net/projects/phpfm/

It crashes on line 4751: `$proc = proc_open($cmd, $descriptorspec, $pipes);`

The complete function:
```
function cmd_proc_open_exec($cmd, &$stdout, &$stderr) {
    $tmp_dir = ini_get('session.save_path') ? ini_get('session.save_path') : sys_get_temp_dir();
    $outfile = tempnam($tmp_dir,"cmd");
    $errfile = tempnam($tmp_dir,"cmd");
    $descriptorspec = array(
        0 => array("pipe", "r"),
        1 => array("file", $outfile, "w"),
        2 => array("file", $errfile, "w")
    );
    $proc = proc_open($cmd, $descriptorspec, $pipes);
    if (!is_resource($proc)) return 255;
    fclose($pipes[0]);    //Don't really want to give any input
    $exit = proc_close($proc);
    $stdout = file_get_contents($outfile);
    $stderr = file_get_contents($errfile);
    @unlink($outfile);
    @unlink($errfile);
    return $exit;
}
```

**Comment by jakubmisek** [comment]
thank you - it's the `array("file", $outfile, "w")` which is not implemented yet at
https://github.com/peachpiecompiler/peachpie/blob/28f7021819c0514c2fb0bf6743324bdbb24bd3eb/src/Peachpie.Library/Processes.cs#L371

it should be fixed in the next update

**Comment by tverweij** [comment]
Thanks!

**Comment by jakubmisek** [comment]
closing, will be fixed in 1.0.0-preview6, thank you!
